controllers/workflow_controller.go

```diff
@@ -101,6 +103,14 @@ func (r *WorkflowReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 		return ctrl.Result{}, client.IgnoreNotFound(err)
 	}
 
+	statusUpdater := newWorkflowStatusUpdater(workflow)
+	defer func() {
+		updateErr := statusUpdater.close(ctx, r)
```

Comment:
Elsewhere we've only closed the updater if an error is not present; is their preference in causing an Update when an error is present? Wondering if we should revisit all our status updaters.

Reply:
I think this one should be switched to not update on an error. Thanks for pointing that out. I'll push the change